pinning/pinner/dspinner/pin.go

```diff
@@ -291,7 +335,14 @@ func (p *pinner) doPinDirect(ctx context.Context, c cid.Cid, name string) error
 		return err
 	}
 
-	return p.flushPins(ctx, false)
+	err = p.flushPins(ctx, false)
+	if err != nil {
+		return err
+	}
+	if p.rootsProvider != nil {
```

Comment:
Note sure if this is required.
```suggestion
	if p.rootsProvider != nil && p.pinnedProvider == nil {
```

Reply:
not necessary because there is no dag walking I think.